kore/app/prof/Main.hs

```diff
@@ -1,19 +1,103 @@
 module Main (main) where
 
-import Options.Applicative qualified as Options
+import Data.Proxy
+import GHC.IO.Encoding (setLocaleEncoding, utf8)
+import Kore.Util.TSM qualified as TSM
+import Kore.Util.TSM.UnifyTag (UnifyTag)
+import Options.Applicative as Options
+import Prelude.Kore
 import Speedscope.Main qualified as Speedscope
-import Prelude
+import System.IO (hFlush, stdout)
 
 main :: IO ()
 main =
-    Speedscope.main =<< Options.execParser info
+    Options.execParser inf
+        >>= \case
+            Speedscope opts -> Speedscope.main opts
+            TSM opts -> runTsm opts
```

Comment:
I wanted to leave the old behaviour intact to avoid disruptions to uses.
We could switch to using commands and sub-parsing if this is not a concern.